modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/agg/Accumulators.java

```diff
@@ -281,7 +281,10 @@
 
         /** {@inheritDoc} */
         @Override public void add(Object... args) {
-            cnt++;
+            assert F.isEmpty(args) || args.length == 1;
```

Comment:
COUNT could work in two modes: count a whole row (COUNT(*), COUNT(1)), and count a particular column in the row (COUNT(col_name)). In the first case the `args` will be empty, and it's OK, since we are not interested in particular columns, only fact that there is a row matters. In the second case we expect a value derived from the current row, and there must be only one such value for each row.